lib/config/configuration.js

```diff
@@ -25,22 +25,148 @@ var BUILTIN_OPTIONS = {
  * @name Configuration
  */
 function Configuration() {
+    /**
+     * List of the registered (not used) presets.
+     *
+     * @protected
+     * @type {Object}
+     */
     this._presets = {};
+
+    /**
+     * Name of the preset (if used).
+     *
+     * @protected
+     * @type {String|null}
+     */
+    this._presetName = null;
+
+    /**
+     * List of loaded presets.
+     *
+     * @protected
+     * @type {String|null}
+     */
+    this._loadedPresets = [];
+
+    /**
+     * List of rules instances.
+     *
+     * @protected
+     * @type {Object}
+     */
     this._rules = {};
+
+    /**
+     * List of configurated rule instances.
+     *
+     * @protected
+     * @type {Object}
+     */
+    this._ruleSettings = {};
+
+    /**
+     * List of configured rules.
+     *
+     * @protected
+     * @type {Array}
+     */
     this._configuredRules = [];
+
+    /**
+     * List of unsupported rules.
+     *
+     * @protected
+     * @type {Array}
+     */
     this._unsupportedRuleNames = [];
+
+    /**
+     * File extensions that would be checked.
+     *
+     * @protected
+     * @type {Array}
+     */
     this._fileExtensions = ['.js'];
+
+    /**
+     * Exclusion masks.
+     *
+     * @protected
+     * @type {Array}
+     */
     this._excludedFileMasks = [];
+
+    /**
+     * List of existing files that falls under exclusion masks.
+     *
+     * @protected
+     * @type {Array}
+     */
     this._excludedFileMatchers = [];
```

Comment:
I think it should be moved to NodeConfiguration if we gonna use `.jscsignore` to extend with internal ignores.